src/view/CustomersCreateWindowState.java

```diff
@@ -0,0 +1,11 @@
+package view;
+
+import controller.CustomerController;
+
+public class CustomersCreateWindowState implements WindowState {	
+	public void OnSave(String name) {
```

Comment:
Atualmente as classes que implementam o padrão State estão instanciando o controller `CustomerController`. Nunca é uma boa prática instanciar um controller, por dois motivos:

1. Controllers em geral são um Singleton. Na nossa implementação, embora os controllers já tenham implementado o método `getInstance()`, eles não possuem o construtor privado. É por isso que, nesse caso, foi possível chamar o `new CustomerController()`. Mas se o Singleton deste controller tivesse sido implementado por completo (o que inclui tornar o seu construtor privado), não seria possível usar o `new`.
2. Quem deve "contatar" o Controller, em geral, são as Views. Nesse caso, as implementações do State não são Views de fato.

O que podemos fazer (também respeitando a implementação do State em si) é, ao invés de passar `String name` como parâmetro, passar a instância da view que usa estes States.

```java
/* @file: CustomersCreateWindowState.java */

// "stateOwner", nesse caso, é a instância da View que usa este State */
public void onSave(JFrame stateOwner) 
{  
  // Chamamos o método `createCustomer` que é implementado lá na View
  stateOwner.createCustomer();
}
```

E, então, lá na View, na hora de chamar o método `onSave()`, ao invés de passar o `name` nós passamos `this`, que é a instância da View:

```java
/* @file: CustomersWindow.java */

public void createCustomer()
{
  // Esse método é chamado lá no State

  // Aqui pegamos os valores do formulário e salvamos no banco, chamando 
  // o Controller normalmente
}
/// ... restante do código

public void actionPerformed(ActionEvent arg0) 
{
  state.onSave(this);
}
```

Mais informações podem ser encontradas aqui: https://github.com/stanley-java-projects/sistema-gerenciador-bibliotecas/issues/24